components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -105,6 +105,9 @@ public OAuth2AuthorizeRespDTO authorize(OAuth2AuthorizeReqDTO oAuth2AuthorizeReq
                     AuthorizationHandlerManager.getInstance();
             return authzHandlerManager.handleAuthorization(oAuth2AuthorizeReqDTO);
         } catch (Exception e) {
+            if (OAuth2Util.isDiagnosticLogsEnabled()) {
+                OAuth2Util.log(null, "FAILED", "System error occurred.", "authorize-client", null);
```

Comment:
No need to check if enabled, if you only use constant strings and already available values

Reply:
Addressed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1737/commits/f7b766bb97ea3607cbc320c6d7b284f05d5668ba